src/plugins/data/common/es_query/es_query/types.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { IFieldType } from '../../index_patterns';
+
+export interface MinimalIndexPattern {
+  fields: IFieldType[];
```

Comment:
Isn't the dependency on `IFieldType[]` going to be a problem too, whenever this gets split out into a package? I'd imagine you'd need to relocate that as well.

FWIW, if it starts to feel weird having all of this stuff crammed in a `@kbn/es-query` package, someone had proposed creating something like `@kbn/data-types` as a type-specific package just for the data plugin, so that these things could live there instead (and be imported by `@kbn/es-query`)

Neither option feels great, just wanted to mention it as something to consider.

Reply:
Yes, I intend to handle `IFieldType` in a followup PR, before moving both into a package